### DISCUSSION TITLE
"on.stopRecording" is called several times

### DISCUSSION THREAD

**Discussion by visualight** [katspaugh/wavesurfer.js#3012] [Q&A] [answered]
Hello,

When I record a voice message the first time, everything goes smoothly. However, when I record a second and then a third message, the function: `this.record.on('stopRecording', () => {}` is called the number of times I've recorded a voice message before.

Is there an event or a method to authorize the call only once per message sent?

Here's my code:

```
startRecording() {
      console.log('startRecording');
      this.isRecording = true
      this.record.startRecording();
    },
    stopRecording() {
      console.log('stopRecording');
      this.isRecording = false
      this.record.stopRecording();

      this.record.on('stopRecording', () => {
        const blobUrl = this.record.getRecordedUrl();
        fetch(blobUrl).then(response => response.blob()).then(blob => {
          this.uploadAudio(blob);
        });
      });
    },
```

Thank you

**Comment by katspaugh** [comment, marked as answer]
That makes sense though, doesn’t it? You’re subscribing to that event as many times as you call the stop method. Changing `on` to `once` should do it.

**Reply by visualight** [reply]
Thank you ... i starting JS yesterday 😂
